components/wifi_provisioning/src/manager.c

```diff
@@ -1425,47 +1428,55 @@ esp_err_t wifi_prov_mgr_start_provisioning(wifi_prov_security_t security, const
      * thread doesn't interfere with this process */
     prov_ctx->prov_state = WIFI_PROV_STATE_STARTING;
 
-    /* Start Wi-Fi in Station Mode.
-     * This is necessary for scanning to work */
-    // esp_err_t err = esp_wifi_set_mode(WIFI_MODE_STA);
-    // if (err != ESP_OK) {
-    //     ESP_LOGE(TAG, "Failed to set Wi-Fi mode to STA");
-    //     RELEASE_LOCK(prov_ctx_lock);
-    //     return err;
-    // }
-    // err = esp_wifi_start();
-    // if (err != ESP_OK) {
-    //     ESP_LOGE(TAG, "Failed to start Wi-Fi");
-    //     RELEASE_LOCK(prov_ctx_lock);
-    //     return err;
-    // }
-
-    /* Change Wi-Fi storage to RAM temporarily and erase any old
-     * credentials in RAM(i.e. without erasing the copy on NVS). Also
-     * call disconnect to make sure device doesn't remain connected
-     * to the AP whose credentials were present earlier */
-    // wifi_config_t wifi_cfg_empty, wifi_cfg_old;
-    wifi_config_t wifi_cfg_old;
-    // memset(&wifi_cfg_empty, 0, sizeof(wifi_config_t));
-    esp_wifi_get_config(ESP_IF_WIFI_STA, &wifi_cfg_old);
-    // err = esp_wifi_set_storage(WIFI_STORAGE_RAM);
-    // if (err != ESP_OK) {
-    //     ESP_LOGE(TAG, "Failed to set Wi-Fi storage to RAM");
-    //     RELEASE_LOCK(prov_ctx_lock);
-    //     return err;
-    // }
-    // esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_cfg_empty);
-    // if (err != ESP_OK) {
-    //     ESP_LOGE(TAG, "Failed to set empty Wi-Fi credentials");
-    //     RELEASE_LOCK(prov_ctx_lock);
-    //     return err;
-    // }
-    // err = esp_wifi_disconnect();
-    // if (err != ESP_OK) {
-    //     ESP_LOGE(TAG, "Failed to disconnect");
-    //     RELEASE_LOCK(prov_ctx_lock);
-    //     return err;
-    // }
+    #if INCLUDE_SPINDANCE_PROVISIONING_SETUP_CHANGES
+        wifi_config_t wifi_cfg_old = {0};
```

Comment:
I took the liberty of initializing this variable, but it likely has no effect, because this included code has no effect (see PR description).